gadget/params.c

```diff
@@ -321,6 +338,19 @@ create_gadget_parameter_set()
     return ps;
 }
 
+/* Structure to hold local parameter values so they can easily be broadcast.
+ * These are for submodules where we do the initialization right here and so don't want to use All.*/
```

Comment:
It is a bit strange initializing modules in params.c. params.c should be just parameter parsing I think. 

I think you were trying to identify per-module items in All, and split them out. If the goal is to modulize All, then probably directly do that instead?

Reply:
That's right. I needed somewhere to store the per-module config locations, since I don't have All to do it anymore. I could just declare them as extern in the per-module headers, in retrospect. Then I'd move the initialisation of ImportBufferBoost out of params as well.

Reply:
I want to avoid having to do a BCast on multiple structures, and I also want to avoid tons of extern globals. But then I agree I also don't really want to do all the init calls from params.

What about a set_*_params() function in the submodule, called from params? And then the init() is separate and called later?

Reply:
Since paramset is part of utils module, I think it is the lesser evil to have every module depend on the paramset.

The module can have a function to register params, and another function that loads param values (which will broadcast). Then the loaded param values can be a per-module local struct.

This does mean gradually we will move params.c into submodules inside libgadget piece by piece.

Reply:
I dunno. That feels a bit like overkill for now. Also it makes it much harder to work out what the parameters of the simulation actually are: right now we just need to open params.c which is easy for the user (although documentation would also be good). 

To be honest, I'm not sure that leaving All intact isn't the best thing to do after all. It at least makes sorting out the multiprocessing simple.